NetCoreRestApi/DataLayer/Repositories/IRepository.cs

```diff
@@ -1,14 +1,14 @@
-﻿using DataLayer.Entities;
+﻿using DataLayer.Models;
 using System.Linq;
 
 namespace DataLayer.Repositories
 {
-    public interface IRepository<T> where T : BaseEntity
+    public interface IRepository<T> where T : BaseModel
     {
-        IQueryable<T> GetAll();
         T GetById(int id);
-        void Create(T entity);
-        void Update(T entity);
-        void Delete(int id);
+        IQueryable<T> GetAll();
+        IQueryable<T> Create(T entity);
+        IQueryable<T> Update(T entity);
+        IQueryable<T> Delete(int id);
```

Comment:
1 Replcae T entity with ICollection entities
2. Update and Delete should return void

Reply:
done